lib/google/gax/api_callable.rb

```diff
@@ -58,33 +63,41 @@ module Gax
     #   * request - Request object
     #   * options - CallOptions object
     #   * block named argument - Proc object for yielding the operation
-    def create_api_call(func, timeout: 60, metadata: {},
+    def create_api_call(func, timeout: 300, metadata: {},
+                        retry_codes: nil, retry_settings: nil,
                         params_extractor: nil, exception_transformer: nil)
       proc do |request, options = nil, block = nil|
         options = CallOptions.new if options.nil?
 
-        options.timeout  = timeout  if options.timeout  == :OPTION_INHERIT
-        options.metadata = metadata if options.metadata == :OPTION_INHERIT
+        options.merge(timeout: timeout, metadata: metadata,
+                      retry_codes: retry_codes,
+                      retry_settings: retry_settings)
 
         if params_extractor
           params = params_extractor.call(request)
           routing_header = params.map { |k, v| "#{k}=#{v}" }.join('&')
           options.metadata['x-goog-request-params'] = routing_header
         end
 
+        retry_manager = RetryManager.new(options)
+
         begin
-          op = func.call(request, deadline: Time.now + options.timeout,
+          op = func.call(request, deadline: retry_manager.deadline,
                                   metadata: options.metadata,
                                   return_op: true)
           res = op.execute
           block.call op if block
           res
-        rescue GRPC::BadStatus => grpc_error
-          error_class = Google::Gax.from_error(grpc_error)
-          error = error_class.new('RPC failed')
-          raise error if exception_transformer.nil?
-          exception_transformer.call error
         rescue StandardError => error
+          if retry_manager.retry?(error)
+            retry_manager.delay!
+            retry
```

Comment:
What is `retry` here?

Reply:
It replays the block, starting at the `begin` a few lines up. This is how the API call is made again.

Reply:
doh, keyword gotcha.